.github/pull_request_template.md

```diff
@@ -0,0 +1,30 @@
+<!-- Provide a general summary of your changes in the Title above -->
+
+### Description
+<!--
+Describe your changes in detail.
+For contribution process guide, look into CONTRIBUTING.md
+
+If this is a relatively large or complex change:
+ - BEFORE creating a PR, find existing or start a new discussion on Issues tab,
+ - if the discussion is concluded, go ahead with this PR,
+ - perhaps describe what alternatives you considered.
+
+If this PR references or fixes an open issue, please link it here
+    using "Ref. #<number>" or "Fixes: #<number>".
+-->
+
+### Checklist
+<!--- Put an 'x' in the boxes that are checked -->
+
+- [ ] Code compiles without errors locally
+- [ ] All tests pass locally
+- [ ] CI workflows executes properly, incl. workflows not executed per PR (e.g. Nightly)
```

Comment:
This implies that the two above are correct too, so they are not needed. Also, this might be a bit too much to require running nightly tests before creating ga PR?

Reply:
heh, yes and no.

1. Running build and tests locally mean you won't spam our CI unnecessarily 😉 
2. You may have a different test environemnt, and that's added bonus + we could potentially find a gap in our docs or e.g. CMake (like [this one](https://github.com/oneapi-src/unified-memory-framework/issues/252))

As for executing Nightly, I agree this might be an overkill, but we tend to forget about it and we don't check this and had at least a few issues on this repo already (like right now, with OOM tests for memspace). That said, I moved this part to optional :)